README.md

```diff
@@ -43,6 +44,33 @@ A
   c: 4
 ```
 
+The macro also supports constructors for named tuples with default values; e.g.
+
+```julia
+julia> MyNT = @with_kw (x = 1, y = "foo", z = :(bar))
+(::#5) (generic function with 2 methods)
+
+julia> MyNT()
+(x = 1, y = "foo", z = :bar)
+```
+
+These constructors can be used as bona fide constructors; e.g. 
+
+```julia
+julia> MyNT(x = 2)
+(x = 2, y = "foo", z = :bar)
+
+julia> MyNT(z = x -> x^3)
+(x = 1, y = "foo", z = #8)
+```
+
+These named tuples can be unpacked exactly like other decorated `struct`s (see below). 
+
+> NOTE: The main caveat is not to reuse a symbol across different assignments; i.e. `MyNT = @with_kw (x = y, c = x)` will yield 
+`MyNT() = (x = foo, c = foo)`, where `foo` is the initial value of `y`. In v0.6, assignments of the kind `f = f` are not supported. 
+
+In v0.6, `NamedTuples.jl` is a prerequisite for named tuple support. In v0.7, they are included in base Julia. 
```

Comment:
Say that NameTuples need to be imported/used.

Reply:
Done.